Bio/SeqUtils/MeltingTemp.py

```diff
@@ -666,7 +674,13 @@ def Tm_Wallace(seq, check=True, strict=True):
         48.0
 
     """
-    seq = str(seq)
+    if isinstance(seq, SeqRecord):
+        seq = str(seq.seq)
+    elif isinstance(seq, Seq._SeqAbstractBaseClass):
+        seq = str(seq)
+    elif not isinstance(seq, str):
+        raise ValueError("Invalid input, must be str, Seq or SeqRecord")
+
     if check:
         seq = _check(seq, "Tm_Wallace")
 
```

Comment:
I haven't tested the code below, but wouldn't it be faster to go over the sequence once and count each of the letters? Because now each call to `seq.count` will cause a separate iteration over the sequence.

If you go over the sequence once, there is the added advantage of not having to check the sequence beforehand. And you won't need to convert lowercase to uppercase.

Also, I would suggest to call `bytes` on the sequence and work with that, to avoid having to convert the `bytes` object stored internally in a `Seq` object to a string object.

Reply:
_I'm assuming this is referencing line 689 in the current commit._
I'm guessing due to how that code is written, the specific breakdown of the time in each part of the `sum` call appears to be opaque to my profiler. However, the builtin sum function accounts for nearly half of the time spent in this function when checks are turned off, so there's a good chance that that is the main bottleneck of the function right now.

However, I don't think that line hasn't changed in the current commit, so improvements there could probably go in a separate PR